### PR TITLE
fix: use https for network calls to APIs

### PR DESCRIPTION
**Pull request by ChristianMurphy**
https://issues.jasig.org/browse/WPT-88

### CODE REVIEW

File: src/test/resources/2502265.xml [hunk 1]
```diff
@@ -42,28 +42,28 @@
             <title>Yahoo! Weather</title>
             <width>142</width>
             <height>18</height>
-            <link>http://weather.yahoo.com</link>
-            <url>http://l.yimg.com/a/i/brand/purplelogo//uh/us/news-wea.gif</url>
+            <link>https://weather.yahoo.com</link>
+            <url>https://l.yimg.com/a/i/brand/purplelogo//uh/us/news-wea.gif</url>
         </image>
```
Comment (bjagg) [inline src/test/resources/2502265.xml]: Test updated URLs. The above does not resolve.

Comment (ChristianMurphy) [reply src/test/resources/2502265.xml]: The image is the old logo, it didn't get migrated to the new server.
The new Yahoo weather icon is https://s.yimg.com/ge/new/WEATHER_ICON.png

**Comment by drewwills** [comment]
I'm a bit confused by this PR (though I'm happy to have it, if it's valuable).  I'm not aware of anything that this PR fixes -- are you guys?

AFAIK, the issue reported by Paul is here:  https://github.com/Jasig/WeatherPortlet/blob/master/src/main/resources/org/jasig/portlet/weather/dao/yahoo/YahooWeatherParsingServiceImpl.groovy#L84

File: src/main/resources/org/jasig/portlet/weather/dao/yahoo/YahooWeatherParsingServiceImpl.groovy [hunk 2]
```diff
@@ -81,7 +81,7 @@ class YahooWeatherParsingServiceImpl implements IYahooWeatherParsingService {
             forecast.setHighTemperature(f.@high.toInteger())
             forecast.setLowTemperature(f.@low.toInteger())
             forecast.setCondition(f.@text.toString())
-            forecast.setImgUrl("http://l.yimg.com/a/i/us/we/52/" + f.@code.toString() + ".gif")
+            forecast.setImgUrl("https://s.yimg.com/zz/combo?/a/i/us/we/52/" + f.@code.toString() + ".gif")
```
Comment (drewwills) [inline src/main/resources/org/jasig/portlet/weather/dao/yahoo/YahooWeatherParsingServiceImpl.groovy]: Just FYI @asgrant.  (Was fixed earlier, but fix was incomplete.)

**Pull request merged**